 Per Peter Tribble, Setting __STDC_VERSION__ to a nonsense value won't lead to a successful outcome.
See also https://github.com/numpy/numpy/issues/25366

--- numpy-2.0.2/numpy/_core/meson.build.orig
+++ numpy-2.0.2/numpy/_core/meson.build
@@ -667,7 +667,6 @@
 
 # Same as NPY_CXX_FLAGS (TODO: extend for what ccompiler_opt adds)
 cpp_args_common = c_args_common + [
-  '-D__STDC_VERSION__=0',  # for compatibility with C headers
 ]
 if cc.get_argument_syntax() != 'msvc'
   cpp_args_common += [
